pkg/kubelet/kubelet.go

```diff
@@ -3076,3 +3118,23 @@ func (kl *Kubelet) warnCgroupV1Usage() {
 	}
 	metrics.CgroupVersion.Set(float64(cgroupVersion))
 }
+
+// staticPodRegistration ensures that all static pods are registered to the apiserver
+// before node became ready.
+func (kl *Kubelet) staticPodsRegistration() error {
+	// kubelet running in standalone mode does not register static pods to the apiserver.
+	if kl.kubeClient == nil {
+		return nil
+	}
+	// Check if initial static pod registration has already been completed.
+	if kl.staticPodsRegistered {
+		return nil
+	}
+	allStaticPods, allMirrorPods := kl.podManager.GetStaticPodsAndMirrorPods()
+	if len(allStaticPods) == len(allMirrorPods) {
```

Comment:
does this guarantees all static pods are mirror or can be scenarios then len is the same but the content is different?

Reply:
yeah. If you look at `GetStaticPodsAndMirrorPods` func, we only fetch static pods and its corresponding mirror pods.